src/igr-tests/igr_functions

```diff
@@ -0,0 +1,232 @@
+# Included from run-test.sh files.
+# shellcheck disable=SC2034,SC2148 # Disable some false warnings/errors.
+
+### This file contain basic variables for running Integration tests of
+### Dinit.
+
+## Basic functions
+# According to POSIX, echo has some unspecified behavior in some cases, for example
+# when its first argument is "-n" or backslash ("\"). We prefer to rely on
+# POSIX documented things.
+# So we replace shell built-in echo with a printf based function.
+# For more info see http://www.etalabs.net/sh_tricks.html
+echo() {
+    IFS=" " printf %s\\n "$*"
+}
+error() {
+    >&2 echo "${TEST_NAME:-}: Error: $*"
+    # Built-in sub_error
+    if [ -n "${2:-}" ]; then
+        >&2 echo " ... $2"
+    fi
+    if [ -n "${STAGE:-}" ]; then
+        echo "${TEST_NAME:-}: Failed at stage $STAGE."
+    fi
+    if [ -e "${SOCKET:-}" ]; then
+        stop_dinit # A dinit instance is running, stopping...
+    fi
+    exit 1
+}
+warning() {
+    echo
+    >&2 echo "${TEST_NAME:-}: Warning: $1"
+    # Built-in sub_warning
+    if [ -n "${2:-}" ]; then
+        >&2 echo " ... $2"
+    fi
+    echo
+}
+
+## Executable path resolvers functions
+# These functions return 0 and set suitable variable when program found.
+# And return 1 on failure.
+__find_dinit() {
+    if [ -z "${DINIT:-}" ]; then
+        if [ -x "$EXEC_PREFIX/dinit" ]; then
+            export DINIT="$EXEC_PREFIX/dinit"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+__find_dinitctl() {
+    if [ -z "${DINITCTL:-}" ]; then
+        if [ -x "$EXEC_PREFIX/dinitctl" ]; then
+            export DINITCTL="$EXEC_PREFIX/dinitctl"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+__find_dinitcheck() {
+    if [ -z "${DINITCHECK:-}" ]; then
+        if [ -x "$EXEC_PREFIX/dinitcheck" ]; then
+            export DINITCHECK="$EXEC_PREFIX/dinitcheck"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+__find_dinitmonitor() {
+    if [ -z "${DINITMONITOR:-}" ]; then
+        if [ -x "$EXEC_PREFIX/dinit-monitor" ]; then
+            export DINITMONITOR="$EXEC_PREFIX/dinit-monitor"
+        else
+            return 1
+        fi
+    fi
+    return 0
+}
+
+## Change directory to run-test.sh location
+cd "$(dirname "$0")" || error "Can't change directory to script location."
+
+## Prepare $IGR_OUTPUT
+TEST_NAME="${PWD##*/}"
+if [ -z "$TEST_NAME" ]; then error "Failed to guess test name."; fi
+if [ -n "${MAKE:-}" ]; then
```

Comment:
Rather than a specific `MAKE` variable just check if `IGR_OUTPUT` is not already set. If it's not set you can assume it's a make-based build (or someone running the script directly, which I do when I am trying to solve a failure in a single test).